### PR TITLE
Implemented D3DShaderErrorInfo and dynamic reloading of DirectX shaders

### PR DESCRIPTION
**Pull request by Gogsi**
Fix for issue #76.
A slightly better version of #77 as it actually provides the error message.
Also created a D3DShaderErrorInfo struct to resemble the OpenGL shader class.

### CODE REVIEW

File: Sparky-core/src/sp/platform/directx/DXShader.cpp [hunk 1]
```diff
@@ -30,28 +54,37 @@ namespace sp { namespace graphics { namespace API {
 
 	void D3DShader::Load(const String& source)
 	{
-		m_Data.vs = Compile(source, "vs_4_0", "VSMain");
+		D3DShaderErrorInfo info;
```
Comment (Ilohama) [inline Sparky-core/src/sp/platform/directx/DXShader.cpp:57]: This shouldn't be like this but like the original so :

```
m_Data.vs = Compile(source, "vs_4_0", "VSMain",info);
if (!m_Data.vs)
    SP_ERROR(info.message);
```

should be :

```
m_Data.vs = Compile(source, "vs_4_0", "VSMain",info);
SP_ASSERT(m_Data.vs, info.profile, info.message);
```

**Comment by TheCherno** [comment]
I'm assuming you've tested this and it works fine?

**Comment by Gogsi** [comment]
Yes I have tested it and it works for both renderers

**Comment by TheCherno** [comment]
Could you update this to resolve conflicts with the latest revision? Thanks.

**Comment by Ilohama** [comment]
As a newest pull request has been merged, this one can be closed.

**Pull request closed without merging**